desktop/logger/index.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Credit: Modified from `winston-namespace` by @SetaSouto:
+ * 	https://github.com/SetaSouto/winston-namespace
+ *
+ * Key changes:
+ *
+ * 1. Use a custom base configuration to avoid overriding the default configuration when
+ * initializing each logger object. This maintains the same logger initialization API
+ * as the prior logging solution (`debug` module).
+ *
+ * 2. Maintain prior log filtering behavior with the `NODE_ENV` environment variable
+ * (instead of the `LOG_NAMESPACES` variable used in winston-namespace).
+ */
+
+/**
+ * External dependencies
+ */
+const { app } = require('electron');
+const path = require('path');
+const { createLogger, format, transports } = require('winston');
+
+/**
+ * Internal dependencies
+ */
+const namespaces = require('./namespaces');
+
+/**
+ * Module variables
+ */
+const maxFiles = 3;
+const maxsize = 15000000;
+
+module.exports = (namespace, options) => {
+  if (!options || typeof options !== 'object') {
+    options = {};
+  }
+
+  const formatMessageWithMeta = (info) => {
+    const args = info[Symbol.for('splat')];
+    if (args) {
+      if (args instanceof Array && args.length && !args[0]) {
+        return info;
+      }
+      info.message = info.message + JSON.stringify(...args);
+    }
+    return info;
+  };
+
+  const baseformat = format.combine(
+    format.timestamp({
+      format: 'YYYY-MM-DD HH:mm:ss.SSS',
+    }),
+    format.errors({ stack: true }),
+    format((info, opts) => formatMessageWithMeta(info, opts))(),
+    format.printf((info) => {
+      const { timestamp, level, message } = info;
+      const stack = info.stack ? `\n${info.stack}` : '';
+      return `[${timestamp}] [${namespace}] [${level}] ${message}` + stack;
+    })
+  );
+
+  const logPath = path.join(
+    (app && app.getPath('appData')) || '',
+    'logs',
+    'simplenote-main.log'
+  );
+
+  const baseOptions = {
+    level: process.env.LOG_LEVEL || 'silly',
```

Comment:
on second thought, by default to `silly` we're saying that customers will by default have `silly` as their log level and any attempt to change that will require that they first quit the application and figure out how to start it again with an `ENV` variable set.

would it do us well to default to something like `warn` instead?

Reply:
Makes sense.